dace/transformation/dataflow/bank_split.py

```diff
@@ -0,0 +1,195 @@
+    split_array_info to [2, 2] and applies the transformation, it will split 
+    equally in all dimensions.
+    Therefore A[0] will be copied to B[0:50, 0:50], A[1] to B[0:50, 50:100], A[2] to B[50:100, 0:50] and
+    A[3] to B[50:100, 50:100].
+
+    Note that simply reversing the AccessNodes for the arrays in the above examples would
+    have lead to the inverse operation, i.e. the gather would become a distribute and 
+    the other way around.
+    """
+
+    _src_node = nd.AccessNode("")
+    _dst_node = nd.AccessNode("")
+
+    # dtype=List[int]
+    split_array_info = properties.Property(
+        dtype=List,
+        default=None,
+        allow_none=True,
+        desc="Describes how many times this array is split in each dimension, "
+        "where the k-th number describes how many times dimension k is split. "
+        "If the k-th number is 1 this means that the array is not split in "
+        "the k-th dimension at all. "
+        "If None, then the transform will split the first dimension exactly shape[0] times."
+    )
+
+    default_to_storage = properties.Property(
+        dtype=dtypes.StorageType,
+        default=dtypes.StorageType.CPU_Heap,
+        allow_none=False,
+        desc=
+        "The storage type of involved arrays will be set to the value of this property if "
+        "they have Default storage type. ")
+
+    def _get_split_size(self, virtual_shape: Iterable,
+                        split_count: List[int]) -> List[int]:
+        """
+        :return: the shape of a part-array on one HBMbank
+        """
+        new_shape_list = []
+        for d in range(len(virtual_shape)):
+            if split_count[d] != 1:
+                new_shape_list.append(virtual_shape[d] // split_count[d])
+            else:
+                new_shape_list.append(virtual_shape[d])
+        return new_shape_list
+
+    @staticmethod
+    def can_be_applied(graph: Union[SDFG, SDFGState],
+                       candidate: Dict['PatternNode', int], expr_index: int,
+                       sdfg: SDFG, strict: bool) -> bool:
+        src = graph.nodes()[candidate[BankSplit._src_node]]
+        dst = graph.nodes()[candidate[BankSplit._dst_node]]
+        src_array = sdfg.arrays[src.data]
+        dst_array = sdfg.arrays[dst.data]
+
+        plain_array = lambda array: isinstance(
+            array, data.Array) and not isinstance(array, data.View)
+
+        if not plain_array(src_array):
+            return False
+        if not plain_array(dst_array):
+            return False
+
+        # same dimensions means HBM-array needs 1 dimension more
+        collect_src = len(src_array.shape) - 1 == len(dst_array.shape)
+        distribute_dst = len(src_array.shape) + 1 == len(dst_array.shape)
+        if collect_src:
+            try:
+                tmp = symbolic.evaluate(src_array.shape[0], sdfg.constants)
+            except TypeError:
+                return False
```

Comment:
Sorry I gave you a suboptimal solution, this can just be:

```suggestion
        if collect_src and symbolic.issymbolic(src_array.shape[0], sdfg.constants):
            return False
```